app/controllers/ops_controller/settings/rhn.rb

```diff
@@ -256,11 +258,12 @@ def rhn_validate
     rhn_load_session
     rhn_fire_available_organizations do |organizations|
       if 'rhn_satellite6' == @edit[:new][:register_to]
+        # Hash of display names to key names
         @edit[:organizations] = organizations
         if @edit[:organizations].length == 1
-          @edit[:new][:customer_org] = @edit[:organizations].first
+          @edit[:new][:customer_org_display], @edit[:new][:customer_org] = @edit[:organizations].first
         else
-          @edit[:new][:customer_org] = nil unless @edit[:organizations].include?(@edit[:new][:customer_org])
+          @edit[:new][:customer_org] = nil unless @edit[:organizations].key?(@edit[:new][:customer_org_display])
```

Comment:
So this is saying "Set `customer_org` to nil unless the current org came back in the organizations list", meaning we are editing, which means it would get pulled from the database.